lib/auth/vnetconfig/v1/service.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package vnetconfig
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/gravitational/teleport/api/gen/proto/go/teleport/vnet/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/services/local"
+)
+
+type Service struct {
+	// Opting out of forward compatibility, this service must implement all service methods.
+	vnet.UnsafeVnetConfigServiceServer
+
+	storage    *local.VnetConfigService
```

Comment:
this is one area where I'm not really sure what we want to do going forward

we could have an interface here but I don't really see why, there is only one storage layer implementation that we'll use, is it to break the dependency api -> storage dependency?

some other types use that same interface and implement it with both the storage layer and an api client. Just to take the most recent example, the CrownJewel service uses this interface in the gRPC api layer to abstract the storage layer, and also makes this gRPC client wrapper implement that same interface

https://github.com/gravitational/teleport/blob/127184f0a82932830c28fd277f24765b0cef8711/lib/auth/crownjewel/crownjewelv1/service.go#L40

https://github.com/gravitational/teleport/blob/127184f0a82932830c28fd277f24765b0cef8711/lib/auth/authclient/clt.go#L639

I do see why it's useful to have `authclient.Client.<SomeService>Client` return an interface rather than a concrete type, so that you can mock `authclient.Client` and have it return a mocked `<SomeService>Client`. What I don't see is why you should need the storage layer to implement that same interface, but maybe it has something to do with the cache needing an interface too and it's just easier to have a single common interface? I just haven't dug that deep yet. Would be nice to have some guidance on this in RFD153 I just haven't figured it out

Reply:
The interest I see here is to be able to mock the underlying storage and test error handling.
Usually we don't do it :disappointed:

Reply:
alright, switch to interface and added tests for storage-layer errors